lms/djangoapps/grades/signals/handlers.py

```diff
@@ -144,6 +145,8 @@ def score_published_handler(sender, block, user, raw_earned, raw_possible, only_
             modified=score_modified_time,
             score_db_table=ScoreDatabaseTableEnum.courseware_student_module,
         )
+        if isinstance(block, ScorableXBlockMixin):
+            block.set_score(Score(raw_earned=raw_earned, raw_possible=raw_possible))
```

Comment:
Also, in your integration testing, you'll need to verify whether this call to `set_score` actually updates the database.  If not, there may need to be a call to `block.save` or something.